pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -4,20 +4,44 @@ import (
 	"fmt"
 	"reflect"
 
+	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/mount"
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 	"github.com/openshift/odo/pkg/lclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/pkg/errors"
 )
 
 // ComponentExists checks if Docker containers labeled with the specified component name exists
 func ComponentExists(client lclient.Client, name string) bool {
+	containers := GetComponentContainers(client, name)
+	return len(containers) != 0
+}
+
+// GetComponentContainers returns a list of the running component containers
+func GetComponentContainers(client lclient.Client, name string) (containers []types.Container) {
 	containerList, err := client.GetContainerList()
 	if err != nil {
-		return false
+		return
 	}
-	containers := client.GetContainersByComponent(name, containerList)
-	return len(containers) != 0
+	containers = client.GetContainersByComponent(name, containerList)
+
+	return
+}
+
+// GetContainerIDForAlias returns the container id for the devfile alias from a list of containers
+func GetContainerIDForAlias(containers []types.Container, componentName string) string {
```

Comment:
Could you add some unit tests for this function? And maybe change `componentName` to something else (as it's used elsewhere in the odo code to refer to the odo component name). Maybe `aliasName`?